File: adventofcode-2016/src/test/java/nl/mvdr/adventofcode/adventofcode2016/day22/Part2Test.java
package nl.mvdr.adventofcode.adventofcode2016.day22;

import java.util.stream.Stream;

import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

class Part2Test extends SolverTest<Part2> {
    
    Part2Test() {
        super(Part2.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(Arguments.of("7", "example-day22.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day07/Part2Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day07

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part2Test: FunctionSolverTest<Long>(::solvePart2) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("40", "example-day07.txt")
            )
        }
    }
}